Mathlib/Algebra/Star/Unitary.lean

```diff
@@ -161,6 +161,38 @@ lemma _root_.isStarNormal_of_mem_unitary {u : R} (hu : u ∈ unitary R) : IsStar
 
 end Monoid
 
+end unitary
+
+section Group
+
+variable {G : Type*} [Group G] [StarMul G]
+
+theorem unitary.inv_mem {g : G} (hg : g ∈ unitary G) : g⁻¹ ∈ unitary G := by
+  simp_rw [unitary.mem_iff, star_inv, ← mul_inv_rev, inv_eq_one] at *
+  exact hg.symm
+
+variable (G) in
+/-- `unitary` as a `Subgroup` of a group.
+
+Note the group structure on this type is not defeq to the one on `unitary`. -/
```

Comment:
I think the only case where one considers unitary elements in a group is in the groups of units of a monoid (I'd be interested to see another example though!), so we should mention it as the use case here.
```suggestion
/-- `unitary` as a `Subgroup` of a group.

Note the group structure on this type is not defeq to the one on `unitary`.
This situation naturally arises when considering the unitary elements as a 
subgroup of the group of units of a star monoid. -/
```

Reply:
> I'd be interested to see another example though

Do module isomorphisms have a star operator? Another example I can think of are things which are morally units of a monoid, but happen to not actually be written with `Units`.